ci: rename `material-unit-tests` job to `components-repo-unit-tests` job

devversion opened the pull request:
We rename the `material-unit-tests` job to `components-repo-unit-tests`
because the job runs all unit tests found in the Angular Components repository.
This includes the Angular CDK, Angular Material and more. Also the repository has
been renamed from `angular/material2` to `angular/components` in the past.

gkalpak commented on .circleci/config.yml:
```diff
@@ -27,10 +27,10 @@ var_4: &cache_key_fallback v3-angular-node-10.16-
-# also update the SHA for the "MATERIAL_REPO_COMMIT" environment variable.
-var_5: &material_unit_tests_cache_key v5-angular-material-97a7e2babbccd3dc58e7b3364004e45ed5bd9968
-var_6: &material_unit_tests_cache_key_fallback v5-angular-material-
+# Cache key for the `components_repo_unit_tests` job. **Note** when updating the SHA in the
```
```suggestion
# Cache key for the `components-repo-unit-tests` job. **Note** when updating the SHA in the
```

gkalpak commented on .circleci/config.yml:
```diff
@@ -27,10 +27,10 @@ var_4: &cache_key_fallback v3-angular-node-10.16-
+# Cache key for the `components_repo_unit_tests` job. **Note** when updating the SHA in the
+# cache keys also update the SHA for the "COMPONENTS_REPO_COMMIT" environment variable.
+var_5: &components_repo_unit_tests_cache_key v5-angular-components-97a7e2babbccd3dc58e7b3364004e45ed5bd9968
+var_6: &components_repo_unit_tests_cache_key_fallback v5-angular-components-
```
Given that `COMPONENT_REPO_TMP_DIR` has changed, I believe you should also update `v5` to `v6`.

devversion replied:
I think we should be fine since the overall prefix of both cache key and fallback key have changed.

gkalpak replied:
Ooops, right, I missed that 😇

gkalpak reviewed (changes requested):
Thx!
<sub>This has been bothering me for a while :grin:</sub>

devversion committed:
ci: rename `material-unit-tests` job to `components-repo-unit-tests`

We rename the `material-unit-tests` job to `components-repo-unit-tests`
because the job runs all unit tests found in the Angular Components repository.
This includes the Angular CDK, Angular Material and more. Also the repository has
been renamed from `angular/material2` to `angular/components` in the past.

devversion commented:
@gkalpak Yeah 😄 This has been on my list for quite some time. Rebased and addressed your feedback!

gkalpak reviewed (approved):
✨

josephperrott reviewed (approved):
LGTM, global approval

AndrewKushnir commented:
@devversion it looks like this PR now has a conflict with master branch (after merging another PR into master). Could you please rebase this PR and resolve the conflict? Thank you.
